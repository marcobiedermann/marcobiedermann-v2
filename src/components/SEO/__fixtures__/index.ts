import { SEOProps } from '../index';

const seo: SEOProps = {
  description: '',
  lang: 'en',
  meta: [],
  title: 'Page title',
};

export default seo;
