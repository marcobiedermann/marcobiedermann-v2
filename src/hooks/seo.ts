import { graphql, useStaticQuery } from 'gatsby';

export interface SEOQuery {
  site: {
    siteMetadata: {
      title: string;
      description: string;
      author: string;
    };
  };
}

export const useSEO = (): SEOQuery =>
  useStaticQuery<SEOQuery>(
    graphql`
      query {
        site {
          siteMetadata {
            title
            description
            author
          }
        }
      }
    `,
  );
