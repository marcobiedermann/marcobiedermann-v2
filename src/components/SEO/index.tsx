import React, { FC } from 'react';
import { Helmet } from 'react-helmet';
import { useSEO } from '../../hooks/seo';

export type Lang = 'en';
export type Meta = {
  name: string;
  content: string;
};

export interface SEOProps {
  description?: string;
  lang?: Lang;
  meta?: Meta[];
  title?: string;
}

export const SEO: FC<SEOProps> = (props) => {
  const { description, lang = 'en', meta = [], title, ...otherProps } = props;
  const { site } = useSEO();

  const metaDescription = description || site.siteMetadata.description;

  return (
    <Helmet
      htmlAttributes={{
        lang,
      }}
      title={title}
      titleTemplate={`%s | ${site.siteMetadata.title}`}
      meta={[
        {
          name: 'description',
          content: metaDescription,
        },
        {
          property: 'og:title',
          content: title,
        },
        {
          property: 'og:description',
          content: metaDescription,
        },
        {
          property: 'og:type',
          content: 'website',
        },
        {
          name: 'twitter:card',
          content: 'summary',
        },
        {
          name: 'twitter:creator',
          content: site.siteMetadata.author,
        },
        {
          name: 'twitter:title',
          content: title,
        },
        {
          name: 'twitter:description',
          content: metaDescription,
        },
      ].concat(meta)}
      {...otherProps}
    />
  );
};

export default SEO;
