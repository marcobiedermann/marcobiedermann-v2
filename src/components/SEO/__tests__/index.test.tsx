/* eslint-disable import/no-extraneous-dependencies */

import { useStaticQuery } from 'gatsby';
import React from 'react';
import renderer from 'react-test-renderer';
import SEO from '..';
import seo from '../__fixtures__';

const mockedUseStaticQuery = useStaticQuery as jest.Mock;

describe('SEO component', () => {
  beforeEach(() => {
    mockedUseStaticQuery.mockImplementationOnce(() => ({
      site: {
        siteMetadata: {
          title: 'Default Starter',
          description: 'Meta description',
        },
      },
    }));
  });

  it('should render correctly', () => {
    const tree = renderer.create(<SEO {...seo} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
