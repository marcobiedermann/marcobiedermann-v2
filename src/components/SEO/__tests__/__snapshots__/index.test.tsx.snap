// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO component should render correctly 1`] = `null`;
